pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -37,8 +37,8 @@ const (
 type Plugin struct{}
 
 // NewPlugin returns an instance of the authn plugin
-func NewPlugin() plugin.Callbacks {
-	return Plugin{}
+func NewPlugin() *Plugin {
```

Comment:
leave the `plugin.Callbacks` as the return value, then you do not need to export the `Plugin` struct